projects/ngrx.io/content/guide/store/architecture.md

```diff
@@ -1 +1,418 @@
+
+### Dispatching an action
+Actions are being *dispatched* to reducer functions. What does that mean? Normally when you set up NGRX you do so be setting up your store like this:
+
+<code-example header="app.module.ts and products.reducer.ts">
+// products.reducer.ts
+export function productsReducer(state = [], action) {
+  switch(action.type) {
+    case '[Products] add':
+      return [...state, action.payload]
+    default:
+      return state;
+  }
+}
+
+// app.module.ts
+StoreModule.forRoot({
+  products: productsReducer
+})
+</code-example>
+
+When you do that you say that `products` is a thing that will exist in a store and it's state is something we can either *read* or *change*. The way we *read* the value is by subscribing to it using the `Store` service inside of component usually like so:
+
+<code-example header="app.module.ts">
+constructor(private store: Store<AppState>) {
+  this.store.pipe(
+    select('products')
+  ).subscribe(products => console.log('products', products));
+}
+</code-example>
+
+The way we *change* we change our data in the store is by calling the same `Store` service and its `dispatch()` method. The `dispatch()` method will *dispatch*, i.e send an action to the `store` and try to match that to a reducer and specifically a matching case in a reducer. So if we in our component where to do this:
+
+<code-example header="products-list.component.ts">
+add() {
+  this.store.dispatch({ 
+    type: '[Products] add', 
+    { id: 1, name: this.name } 
+  });
+}
+</code-example>
+
+Then we would match the case in our `products.reducer.ts`:
+
+<code-example header="products.reducer.ts">
+case '[Products] add':
+  return [...state, action.payload]
+</code-example>
+
+The end result would be that our `products` store property would now contain one more product.
+
+### When an action fails
+This all seems good right. Set up a property, associate to a reducer, dispatch an action and our state changes. All is good or? All is good UNTIL you mispel your `action.type`. Consider the following code below:
+
+<code-example header="Reducer example">
+function reducer(state = 0, action) {
+  switch(action.type) {
+    case 'INCREMENT':
+      return state + 1;
+    default:
+      return state;
+  }
+}
+// this WON'T work as intended 
+reducer({ type: 'INREMENT' })
+
+// this WILL work, cause it's correctly spelled
+reducer({ type: 'INCREMENT' })
+</code-example>
+
+Above we are showcasing how actions are used, namely that they are objects being sent to a reducer functions that lets them jump through a `switch` construct. We are also demonstrating how the spelling of your action matters. If we spell it `INREMENT` it wont hit the intended case but if we correctly spell it `INCREMENT` it will hit the correct case. 
```

Comment:
~~It's also the payload that could be misspelled, or using the "wrong" payload. Don't know if it's worth adding or not...~~

This is added below 😄